crates/rustc_codegen_spirv/src/builder/builder_methods.rs

```diff
@@ -1974,6 +1975,128 @@ impl<'a, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'tcx> {
             // needing to materialize `&core::panic::Location` or `format_args!`.
             self.abort();
             self.undef(result_type)
+        } else if self.internal_buffer_load_id.borrow().contains(&llfn_def) {
+            // simple structs are returned as values, complex ones are returned as out parameters
```

Comment:
Is there a good way to detect this other then going, "well I see we have one extra out-parameter now"?

Reply:
`extern "unadjusted"` resolved this.